drivers/i3c/i3c_stm32.c

```diff
@@ -0,0 +1,2129 @@
+		return -EIO;
+	}
+
+	/* Satisfying I3C start timing min timing will statisfy the rest of the conditions */
+
+	if (i2c_bus_freq != 0) {
+		if (i2c_bus_freq > 400000) {
+			/* Mixed bus with I2C FM+ device */
+			free_timing = (uint8_t)ceil(
+				(STM32_I3C_TBUF_FMP_MIN_NS * i3c_clock / 1e9 - 0.5) / 2);
+		} else {
+			/* Mixed bus with I2C FM device */
+			free_timing = (uint8_t)ceil(
+				(STM32_I3C_TBUF_FM_MIN_NS * i3c_clock / 1e9 - 0.5) / 2);
+		}
+	} else {
+		if (config->drv_cfg.dev_list.num_i2c > 0) {
+			enum i3c_bus_mode mode = i3c_bus_mode(&config->drv_cfg.dev_list);
+
+			if (mode == I3C_BUS_MODE_MIXED_FAST) {
+				if (get_i3c_lvr_ic_mode(&config->drv_cfg.dev_list) ==
+				    I3C_LVR_I2C_FM_MODE) {
+					/* Mixed bus with I2C FM device */
+					free_timing = (uint8_t)ceil(
+						(STM32_I3C_TBUF_FM_MIN_NS * i3c_clock / 1e9 - 0.5) /
+						2);
+				} else {
+					/* Mixed bus with I2C FM+ device */
+					free_timing = (uint8_t)ceil(
+						(STM32_I3C_TBUF_FMP_MIN_NS * i3c_clock / 1e9 -
+						 0.5) /
+						2);
+				}
+			} else {
+				return -EINVAL;
+			}
+		} else {
+			/* Pure I3C bus */
+			free_timing =
+				(uint8_t)ceil((STM32_I3C_TCAS_MIN_NS * i3c_clock / 1e9 - 0.5) / 2);
+		}
+	}
+
+	aval = DIV_ROUND_UP(1000ull * i3c_clock, 1000000000ull) - 1;
+
+	LL_I3C_SetDataHoldTime(i3c, LL_I3C_SDA_HOLD_TIME_1_5);
+	LL_I3C_SetFreeTiming(i3c, free_timing);
+	LL_I3C_SetControllerActivityState(i3c, LL_I3C_OWN_ACTIVITY_STATE_0);
+	LL_I3C_SetAvalTiming(i3c, aval);
+
+	LOG_DBG("TimingReg1 = 0x%08x", LL_I3C_GetCtrlBusCharacteristic(i3c));
+
+	return 0;
+}
+
+/* Configures the I3C module in controller mode */
+static int i3c_stm32_configure(const struct device *dev, enum i3c_config_type type, void *cfg)
+{
+	int ret;
+
+	if (type == I3C_CONFIG_TARGET || type == I3C_CONFIG_CUSTOM) {
+		return -EOPNOTSUPP;
+	}
+
+	const struct i3c_stm32_config *config = dev->config;
+	struct i3c_stm32_data *data = dev->data;
+
+	I3C_TypeDef *i3c = config->i3c;
+
+	ret = i3c_stm32_activate(dev);
+	if (ret != 0) {
+		LOG_ERR("Clock and GPIO could not be initialized for the I3C module, err=%d", ret);
+		return ret;
+	}
+
+	/* I3C Initialization */
+	LL_I3C_SetMode(i3c, LL_I3C_MODE_CONTROLLER);
+	LL_I3C_SetDataHoldTime(i3c, LL_I3C_SDA_HOLD_TIME_1_5);
+	LL_I3C_SetControllerActivityState(i3c, LL_I3C_OWN_ACTIVITY_STATE_0);
+
+	ret = i3c_stm32_config_clk_wave(dev);
+	if (ret != 0) {
+		LOG_ERR("TimigReg0 timing could not be calculated, err=%d", ret);
+		return ret;
+	}
+
+	ret = i3c_stm32_config_ctrl_bus_char(dev);
+	if (ret != 0) {
+		LOG_ERR("TimingReg1 timing could not be calculated, err=%d", ret);
+		return ret;
+	}
+
+#ifdef CONFIG_I3C_USE_IBI
+	LL_I3C_EnableHJAck(i3c);
+#endif
+
+	/* Configure FIFO */
+	LL_I3C_SetRxFIFOThreshold(i3c, LL_I3C_RXFIFO_THRESHOLD_1_4);
+	LL_I3C_SetTxFIFOThreshold(i3c, LL_I3C_TXFIFO_THRESHOLD_1_4);
+	LL_I3C_EnableControlFIFO(i3c);
+	LL_I3C_EnableStatusFIFO(i3c);
+
+	/* Configure Controller */
+	LL_I3C_SetOwnDynamicAddress(i3c, 0);
```

Comment:
There shouldn't be a dynamic address of 0. use one of the i3c address apis to pick one (or assign one with a dts prop) if it is a primary controller

Reply:
So... this was just removed? I'm okay with removing it as target mode (nor handoff) is supported by this driver yet.

Reply:
It was never intended for the controller to have a dynamic address to begin with, that's why we decided to remove it